### DISCUSSION TITLE
php warning on product page. prestashop 8.2 php 8.1

### DISCUSSION THREAD

**Discussion by kangn326** [PrestaShop/PrestaShop#37705] [Q&A]
Does anyone know about this bug? and any fix?

 

PHP Warning: Undefined array key "cart_quantity" in /public_html/controllers/front/ProductController.php on line 1329

 

    /**
     * @param array $product
     *
     * @return int
     */
    protected function getRequiredQuantity($product)
    {
        $requiredQuantity = (int) Tools::getValue('quantity_wanted', $this->getProductMinimalQuantity($product));
        if ($requiredQuantity < $product['minimal_quantity']) {
            $requiredQuantity = $product['minimal_quantity'];
        }

        if ($product['cart_quantity'] >= $requiredQuantity) {
            return 0;
        }

        return $requiredQuantity;

**Comment by kpodemski** [comment]
Hello @kangn326 

Is there any specific moment where you encounter this error? Is it only available in the logs or is it displayed in front office of your store?

**Reply by sirius82** [reply]
Hello, I have the same warning with the same configuration. This error displays when debug is enabled. It's just a warning but it should be beter to not have it :-)
Thanks
